### PR TITLE
[stable/acs-engine-autoscaler] Add acs-engine-autoscaler chart

### PR DESCRIPTION
**Pull request by ritazh**
This chart deploys [acs-engine-autoscaler](https://github.com/wbuchwalter/Kubernetes-acs-engine-autoscaler), a node-level autoscaler for Kubernetes for clusters created with acs-engine.

Tested with [image tag 0.3.1](https://hub.docker.com/r/ritazh/kubernetes-acs-engine-autoscaler/) on a v1.6.0 k8s. 

cc @wbuchwalter

### CODE REVIEW

**Comment by ritazh** [comment]
@seanknox Would love to get your feedback on this one. Thanks!!! :bowing_woman:

File: incubator/acs-engine-autoscaler/README.md
Comment (seanknox) [inline incubator/acs-engine-autoscaler/README.md]: Hooray! This will be a great chart addition.

File: incubator/acs-engine-autoscaler/templates/NOTES.txt
Comment (seanknox) [inline incubator/acs-engine-autoscaler/templates/NOTES.txt]: As this chart requires credentials to run, you should output an error if they're missing. Check out https://github.com/kubernetes/charts/blob/master/stable/sumokube/templates/NOTES.txt for one pattern. No point in a user checking pods if core config is missing.

File: incubator/acs-engine-autoscaler/templates/_helpers.tpl
Comment (seanknox) [inline incubator/acs-engine-autoscaler/templates/_helpers.tpl]: Replace this _helpers.tpl with this one: https://github.com/kubernetes/charts/blob/master/incubator/istio/templates/_helpers.tpl

It enables additional patterns we're trying to establish, such as DRY'ing up the label boilerplate with the helper function.

File: incubator/acs-engine-autoscaler/Chart.yaml
Comment (seanknox) [inline incubator/acs-engine-autoscaler/Chart.yaml]: This chart can be in `stable/`. incubator is required only if you're using experimental Kubernetes functionality: alpha APIs, for example.

File: incubator/acs-engine-autoscaler/README.md
Comment (seanknox) [inline incubator/acs-engine-autoscaler/README.md]: Is there any benefit to configuring the Azure deployment name? Wondering if there's any harm to always using a default name.

File: incubator/acs-engine-autoscaler/templates/deployment.yaml
Comment (seanknox) [inline incubator/acs-engine-autoscaler/templates/deployment.yaml]: is the openai label still present in the autoscaler? Just curious.

File: incubator/acs-engine-autoscaler/values.yaml
Comment (seanknox) [inline incubator/acs-engine-autoscaler/values.yaml]: Don't forget to pin image to a specific version tag.

**Review by seanknox — CHANGES_REQUESTED** [review]
Impressive first chart! Thanks for taking this on.

File: incubator/acs-engine-autoscaler/README.md
Comment (wbuchwalter) [reply incubator/acs-engine-autoscaler/README.md]: This is actually the name that was used to deploy the cluster generated with acs-engine via `az group deployment create ...` .  Not a name for future scaling deployments.

File: incubator/acs-engine-autoscaler/templates/deployment.yaml
Comment (wbuchwalter) [reply incubator/acs-engine-autoscaler/templates/deployment.yaml]: Yep!

File: incubator/acs-engine-autoscaler/Chart.yaml
Comment (wbuchwalter) [reply incubator/acs-engine-autoscaler/Chart.yaml]: Should this be changed for "Scales worker nodes within agent pools" ?

File: incubator/acs-engine-autoscaler/README.md
Comment (ritazh) [reply incubator/acs-engine-autoscaler/README.md]: @seanknox This argument is optional. It is here to support users who have provided a custom Azure deployment name, not using the default `azuredeploy` name at the time of running `az group deployment create `

From the `deployment.yaml`:
```
{{- if .Values.acsenginecluster.acsdeployment}}
            - --acs-deployment 
            - {{ .Values.acsenginecluster.acsdeployment }}
```

File: incubator/acs-engine-autoscaler/Chart.yaml
Comment (seanknox) [reply incubator/acs-engine-autoscaler/Chart.yaml]: 👍

Comment (ritazh) [reply incubator/acs-engine-autoscaler/Chart.yaml]: done.

**Comment by ritazh** [comment]
@seanknox I have made the suggested updates. Once approved, I will squash my commits.

File: stable/acs-engine-autoscaler/templates/deployment.yaml
Comment (seanknox) [inline stable/acs-engine-autoscaler/templates/deployment.yaml]: You can replace the manually defined labels with your fancy new `labels.standard` helper function. Example here: https://github.com/kubernetes/charts/blob/master/incubator/istio/templates/addons-grafana-deployment.yaml#L8

File: stable/acs-engine-autoscaler/templates/secrets.yaml
Comment (seanknox) [inline stable/acs-engine-autoscaler/templates/secrets.yaml]: Minor nit: for consistency's sake you should add the labels to all K8s resources managed by Helm (again, using `labels.standard`).

**Comment by seanknox** [comment]
Looking great. Two minor comments and I'll merge—no need to squash unless you want to, as I can do it at merge time.

**Review by seanknox — CHANGES_REQUESTED** [review]

**Review by seanknox — APPROVED** [review]
You rock. Thank you Rita!

**Pull request merged**
